mesonbuild/cmake/interpreter.py

```diff
@@ -664,6 +664,7 @@ def ensure_absolute(x: Path) -> Path:
         commands = []  # type: T.List[T.List[T.Union[str, ConverterTarget]]]
         for curr_cmd in self._raw_target.command:
             assert isinstance(curr_cmd, list)
+            assert curr_cmd[0] != '', "An empty string is not a valid executable"
```

Comment:
Is this supposed to:
- catch programming errors in the cmake module
- tell mypy something about type checking that it cannot infer automatically
- emit user errors for a cmake project that for whatever reason cannot be parsed